### ISSUE TITLE
Git commands are slow

### ISSUE THREAD

**Issue by AlexSchulz98** [Microsoft/vscode#51581]
Issue Type: <b>Performance Issue</b>

git actions like syncing or commiting to my repository (gitlab) are really slow. It only occurs on my laptop. I've already updated to the latest version of VSC and git (2.17.1.windows.2).
At my PC at home it takes about 3 seconds to sync with the repository, here it takes about 30s. 

VS Code version: Code 1.24.0 (6a6e02cef0f2122ee1469765b704faf5d0e0d859, 2018-06-06T17:35:40.560Z)
OS version: Windows_NT x64 10.0.17134

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7500U CPU @ 2.70GHz (4 x 2904)|
|GPU Status|2d_canvas: unavailable_software<br>flash_3d: unavailable_software<br>flash_stage3d: unavailable_software<br>flash_stage3d_baseline: unavailable_software<br>gpu_compositing: unavailable_software<br>multiple_raster_threads: unavailable_off<br>native_gpu_memory_buffers: disabled_software<br>rasterization: unavailable_software<br>video_decode: unavailable_software<br>video_encode: unavailable_software<br>vpx_decode: unavailable_software<br>webgl: unavailable_off<br>webgl2: unavailable_off|
|Memory (System)|7.85GB (3.77GB free)|
|Process Argv|C:\Program Files\Microsoft VS Code\Code.exe|
|Screen Reader|no|
|VM|0%|

</details><details>
<summary>Process Info</summary>

```
CPU %	Mem MB	   PID	Process
    3	   134	  4192	code main
    0	    67	  7704	   shared-process
    0	    76	  8816	   shared-process
    0	   228	  8840	   window (website - Visual Studio Code)
    1	    62	  5232	     extensionHost
    0	     5	  7080	       "C:\Program Files\Git\cmd\git.exe" symbolic-ref --short HEAD
    0	     9	  3384	         console-window-host (Windows internal process)
    0	    38	  7920	       electron_node jsonServerMain.js 
    0	    11	  6648	     electron-crash-reporter
    0	    32	  7220	     terminal
    0	     6	  3448	       winpty-process
    0	     9	  3780	         console-window-host (Windows internal process)
    0	    62	  5056	         C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe
    0	    14	 11608	     watcherService 
    0	     9	  7208	       console-window-host (Windows internal process)
    0	    27	 12836	   gpu-process
```

</details>
<details>
<summary>Workspace Info</summary>

```
|  Window (website - Visual Studio Code)
|    Folder (website): 180 files
|      File types: ts(80) html(27) scss(27) png(14) js(11) json(8) ico(4)
|                  xlf(2) gitignore(1) css(1)
|      Conf files: package.json(1) tsconfig.json(1) tslint.json(1);
```

</details>
<details><summary>Extensions (1)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-language-pack-de|MS-|1.24.3

</details>

**Comment by joaomoreno** [comment]
Interesting. We've been having a couple of other people with the same issue, but they have fixed it by updating to the latest git. Can you uninstall and reinstall the latest Git version? Let's see if that changes anything.

**Comment by AlexSchulz98** [comment]
I just uninstalled and reinstalled git but it did not solve the problem...
Still really slow.

**Comment by joaomoreno** [comment]
Do you have an Antivirus? If so, does it get better if you temporarily disable it?

**Comment by AlexSchulz98** [comment]
Yes, I have eset nod32, but even with this being diabled it took about 30 seconds. So antivirus should not be the problem.
commit and push via command line are really fast (it takes about a second). It's just the vscode options that take forever. 
Same with creating a new branch. it takes about 10s before I can see my new branch at the bottom and the little cloud-icon to push it appears.

**Comment by joaomoreno** [comment]
Try to go through other similar issues, see if you find something which fixes it for you: #3873, #34898, etc

**Comment by joaomoreno** [comment]
Too bad... let me know if you figure it out, I'd be very happy to know! I wish I could reproduce it and investigate it on my machine...
